lib/main.dart

```diff
@@ -281,6 +282,12 @@ class App extends StatefulWidget {
 }
 
 class AppState extends State<App> with SingleTickerProviderStateMixin {
+  @override
+  void dispose() {
+    getIt.get<TradeMonitor>().dispose();
+    super.dispose();
+  }
```

Comment:
I believe this will only get called when closing the app, and if so then it doesn't make sense to dispose anything then, it will already be disposed automatically?

Unless, this is getting called when switching wallets, is that the case? and if so, did we test if there would be a race condition of whether the timers of the new wallet would get registered before this function is called, and then this will remove the timers?